File: SafeDalat_API/SafeDalat_API/Model/DTO/QA/QuestionResponseDTO.cs
namespace SafeDalat_API.Model.DTO.QA
{
    public class QuestionResponseDTO
    {
        public int QuestionId { get; set; }
        public string Content { get; set; } = null!;
        public DateTime CreatedAt { get; set; }

        public int UserId { get; set; }
        public string UserName { get; set; } = null!;

        public string QuestionCategoryName { get; set; } = null!;
        public string? AssignedDepartmentName { get; set; } // Nếu đã được điều phối

        public List<AnswerResponseDTO> Answers { get; set; } = new();
    }
}

File: DalatS/app/src/main/java/com/example/dalats/model/QuestionResponseDTO.java
package com.example.dalats.model;
import com.google.gson.annotations.SerializedName;
import java.io.Serializable;
import java.util.List;

public class QuestionResponseDTO implements Serializable {
    @SerializedName("questionId")
    private int questionId;
    @SerializedName("content")
    private String content;
    @SerializedName("userName")
    private String userName;
    @SerializedName("questionCategoryName")
    private String questionCategoryName;
    @SerializedName("createdAt")
    private String createdAt;
    @SerializedName("answers")
    private List<AnswerResponseDTO> answers;

    public int getQuestionId() { return questionId; }
    public String getContent() { return content; }
    public String getUserName() { return userName; }
    public String getQuestionCategoryName() { return questionCategoryName; }
    public String getCreatedAt() { return createdAt; }
    public List<AnswerResponseDTO> getAnswers() { return answers; }
}